pulp_container/app/downloaders.py

```diff
@@ -57,6 +57,8 @@ async def _run(self, handle_401=True, extra_data=None):
         headers.update(auth_headers)
         # aiohttps does not allow to send auth argument and auth header together
         self.session._default_auth = None
+        if self.download_throttler:
+            await self.download_throttler.acquire()
```

Comment:
I'm not an expert in this area. The interface used here looks like download_throttler was a lock or semaphore, which kind of makes sense.

Reply:
i took it from core https://github.com/pulp/pulpcore/blob/main/pulpcore/download/http.py#L255